crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -104,25 +104,45 @@ impl AuthorityStorePruner {
         checkpoint_number: CheckpointSequenceNumber,
         metrics: Arc<AuthorityStorePruningMetrics>,
         indirect_objects_threshold: usize,
+        enable_pruning_tombstones: bool,
     ) -> anyhow::Result<()> {
         let _scope = monitored_scope("ObjectsLivePruner");
         let mut wb = perpetual_db.objects.batch();
 
-        let mut object_keys_to_prune = vec![];
+        // Collect objects keys that need to be deleted from `transaction_effects`.
+        let mut live_object_keys_to_prune = vec![];
+        let mut object_tombstones_to_prune = vec![];
         for effects in &transaction_effects {
             for (object_id, seq_number) in effects.modified_at_versions() {
-                object_keys_to_prune.push(ObjectKey(object_id, seq_number));
+                live_object_keys_to_prune.push(ObjectKey(object_id, seq_number));
+            }
+
+            // Indirect object is not being used when pruning tombstone is developed. Therefore, to not further
+            // complicate the code, tombstone pruning is only enabled when indirect object is not turned on.
+            // TODO: re-evaluate the need for indirect object.
+            if enable_pruning_tombstones
+                && (indirect_objects_threshold == 0 || indirect_objects_threshold == usize::MAX)
+            {
+                for deleted_object_ref in effects
+                    .all_removed_objects()
```

Comment:
its not obvious to me that this matches the logic in https://github.com/MystenLabs/sui/blob/066673c94744649314e337f0f345036656302f94/crates/sui-core/src/authority/authority_store.rs#L1143-L1160

can you clarify?

Reply:
This got me confused too... The reason I was using `all_removed_objects` is because of here [https://github.com/MystenLabs/sui/blob/61428d142af8a47f3600da543d547d4868a55c94/crates/sui-core/src/authority/authority_store.rs#L1560](url) . After confirming with Xun, looks like we do create tombstones for `unwrap_then_delete` objects. So good catch!

I modified here to include `unwrap_then_delete` objects. I may create another helper function in transaction effects to be used in places where tombstones are used.